<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Rafael Lopes">
<meta name="dcterms.date" content="2023-04-15">

<title>Assessement of US genomic sequence metadata</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="Assessment_metadata_files/libs/clipboard/clipboard.min.js"></script>
<script src="Assessment_metadata_files/libs/quarto-html/quarto.js"></script>
<script src="Assessment_metadata_files/libs/quarto-html/popper.min.js"></script>
<script src="Assessment_metadata_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Assessment_metadata_files/libs/quarto-html/anchor.min.js"></script>
<link href="Assessment_metadata_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Assessment_metadata_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Assessment_metadata_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Assessment_metadata_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Assessment_metadata_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="Assessment_metadata_files/libs/quarto-diagram/mermaid.min.js"></script>
<script src="Assessment_metadata_files/libs/quarto-diagram/mermaid-init.js"></script>
<link href="Assessment_metadata_files/libs/quarto-diagram/mermaid.css" rel="stylesheet">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#genomic-metadata" id="toc-genomic-metadata" class="nav-link active" data-scroll-target="#genomic-metadata">Genomic metadata</a></li>
  <li><a href="#loading-the-global-metadata" id="toc-loading-the-global-metadata" class="nav-link" data-scroll-target="#loading-the-global-metadata">Loading the global metadata</a>
  <ul class="collapse">
  <li><a href="#flow-chart-on-filtering-metada" id="toc-flow-chart-on-filtering-metada" class="nav-link" data-scroll-target="#flow-chart-on-filtering-metada">Flow chart on filtering metada</a></li>
  <li><a href="#filtering-and-categorization" id="toc-filtering-and-categorization" class="nav-link" data-scroll-target="#filtering-and-categorization">Filtering and categorization</a></li>
  <li><a href="#plotting-the-data" id="toc-plotting-the-data" class="nav-link" data-scroll-target="#plotting-the-data">Plotting the data</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Assessement of US genomic sequence metadata</h1>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    Rafael Lopes 
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            Yale University, Department Epidemiology of Microbial Diseases and Public Health Modeling Unit
          </p>
      </div>
    </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 15, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="genomic-metadata" class="level2">
<h2 class="anchored" data-anchor-id="genomic-metadata">Genomic metadata</h2>
<p>From <a href="https://gisaid.org">GISAID</a> is a genomic database maintained from collective uploading by researchers, has been the main platform collecting the genetic and genomic data sequence of SARS-CoV-2 virus. From the metadata of this sequences, we have made this assessment on the information and surveillance sequencing in the US.</p>
</section>
<section id="loading-the-global-metadata" class="level2">
<h2 class="anchored" data-anchor-id="loading-the-global-metadata">Loading the global metadata</h2>
<p>The global metadata from GISAID sizes ~11GB, with information from the purpose, the date of collection of the sample, the date of submission of the sequence into GISAID, the location (the micro spatial level, at US county-level information), the division (the middle spatial level, at US state-level information), the region (the macro spatial level, for the US data, the sub-continental information, North America region)</p>
<section id="flow-chart-on-filtering-metada" class="level3">
<h3 class="anchored" data-anchor-id="flow-chart-on-filtering-metada">Flow chart on filtering metada</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<p>
</p><pre class="mermaid mermaid-js" data-tooltip-selector="#mermaid-tooltip-1">flowchart LR
  A[(Raw Global Metadata)] -- US deposited sequences --&gt; B[(US Metadata)]
  B --&gt; C{Variant categorization}
  C --&gt; D[Count of sequences]
  C --&gt; E[Frequency of sequeces]
</pre>
<div id="mermaid-tooltip-1" class="mermaidTooltip">

</div>
<p></p>
</div>
</div>
</div>
</section>
<section id="filtering-and-categorization" class="level3">
<h3 class="anchored" data-anchor-id="filtering-and-categorization">Filtering and categorization</h3>
<p>We have filtered the metadata to only count for human host sequences and sequences regarding case from the US.</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Loading packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>packs<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"tidyverse"</span>, <span class="st">"vroom"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(packs, require, <span class="at">character.only =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Loading functions</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"Functions/functions.R"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Loading the metadata and states</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>metadata<span class="ot">&lt;-</span>vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="st">"../Data/metadata/metadata_2023-04-09_23-54.tsv.gz"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="do">## States abbreviation</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>states<span class="ot">&lt;-</span>vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="st">"../Data/state_abbreviation.tsv"</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Filtering for the US, human only infections, and formatting date</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>metadata<span class="ot">&lt;-</span>metadata <span class="sc">|&gt;</span> </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(country <span class="sc">==</span> <span class="st">"USA"</span>, </span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>         host <span class="sc">==</span> <span class="st">"Human"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">date =</span> <span class="fu">as.Date</span>(date, <span class="st">"%Y-%m-%d"</span>), </span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>         <span class="do">## Fixing DC</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">division =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(division <span class="sc">==</span> <span class="st">"Washington DC"</span> <span class="sc">~</span> <span class="st">"District of Columbia"</span>, </span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>                              <span class="cn">TRUE</span> <span class="sc">~</span> division)) <span class="sc">|&gt;</span> </span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Filtering out any state name that is not matching 49 contiguous states plus Alaska and Hawaii</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(division <span class="sc">%in%</span> states<span class="sc">$</span>name_state)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>After this, we categorize the sequences by search on their “Pango lineage” columns and sub-setting them as any categories. The choices on categorization were based on the Variants of Concern monitored from the <a href="https://covid.cdc.gov/covid-data-tracker/#variant-proportions">CDC Variant Tracker</a>. From the categories that CDC maintains nowcasting projections, we reduced them to only 7, being them:</p>
<ul>
<li><strong>Omicron BA.1</strong>*: Any lineage with BA.1 pango lineage and/or its child lineages, e.g., BA.1.1, BA.1.2, and so on</li>
<li><strong>Omicron BA.2</strong>*: Any lineage with BA.2 pango lineage and/or its child lineages, e.g., BA.2.1, BA.2.2, and so on</li>
<li><strong>Omicron BA.2.75</strong>*: Any lineage with BA.2 pango lineage and/or its child lineages, e.g., BA.2.75.1, BA.2.75.2, and so on</li>
<li><strong>Omicron BA.3</strong>*: Any lineage with BA.2 pango lineage and/or its child lineages, e.g., BA.3.1, BA.3.2, and so on</li>
<li><strong>Omicron BA.4</strong>*: Any lineage with BA.2 pango lineage and/or its child lineages, e.g., BA.4.1, BA.4.2, and so on</li>
<li><strong>Omicron BA.5</strong>*: Any lineage with BA.2 pango lineage and/or its child lineages, e.g., BA.5.1, BA.5.2, and so on</li>
<li><strong>Omicron XBB</strong>*: Any lineage with BA.2 pango lineage and/or its child lineages, e.g., XBB.1, XBB.2, XBB.1.5, and so on</li>
<li><strong>Other</strong>: Any lineage that were no categorized in the above categories, this is the category of the spare lineages, it has a constant counting and frequency, but always on low numbers.</li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Formatting metadata in a more pratical form</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata <span class="sc">|&gt;</span> </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Dummy date variable</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">copy_date =</span> <span class="fu">as.numeric</span>(date)) <span class="sc">|&gt;</span> </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Choosen lineages to be analyzed</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">voc_cdc =</span> <span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'^BA</span><span class="sc">\\</span><span class="st">.1'</span>, </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'Omicron BA.1*'</span>,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'BA.2.75'</span>, </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'Omicron BA.2.75*'</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'^(?=.*BA.2)(?!.*BA.2.75)'</span>, </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'Omicron BA.2*'</span>,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'BA.3'</span>, </span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'Omicron BA.3*'</span>,</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'BA.4'</span>, </span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'Omicron BA.4*'</span>,</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'BA.5'</span>, </span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'Omicron BA.5*'</span>, </span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'XBB.1.5'</span>, </span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'XBB.1.5*'</span>,</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">grepl</span>(<span class="at">x =</span> pango_lineage, </span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pattern =</span> <span class="st">'XBB.1'</span>, </span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">perl =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">'XBB.1*'</span>,</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>                             <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">'Other'</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>         <span class="do">## Completing dates to avoid missing dates</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>         <span class="fu">complete</span>(date, <span class="fu">nesting</span>(voc_cdc), <span class="at">fill =</span> <span class="fu">list</span>(<span class="at">copy_date =</span> <span class="dv">0</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>         <span class="do">## Putting date at the ending date of the epiweek</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>         <span class="fu">mutate</span>(<span class="at">epiweek =</span> <span class="fu">end.of.epiweek</span>(date))</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Variant sequences counts</span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>variant_count<span class="ot">&lt;-</span>metadata <span class="sc">|&gt;</span> </span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  <span class="do">## First probable sequence of Omicron in the US</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(date <span class="sc">&gt;=</span> <span class="st">"2021-09-01"</span>, <span class="sc">!</span><span class="fu">is.na</span>(division)) <span class="sc">|&gt;</span> </span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  <span class="do">## dummy date column</span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">copy_date =</span> <span class="fu">as.numeric</span>(epiweek)) <span class="sc">|&gt;</span> </span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Grouping by epiweek, state and voc_cdc</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(epiweek, division, voc_cdc) <span class="sc">|&gt;</span> </span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_at</span>(<span class="fu">vars</span>(copy_date), <span class="fu">list</span>(<span class="at">n =</span> sum)) <span class="sc">|&gt;</span></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>n<span class="sc">/</span><span class="fu">sum</span>(n),<span class="dv">2</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">name_states =</span> division)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plotting-the-data" class="level3">
<h3 class="anchored" data-anchor-id="plotting-the-data">Plotting the data</h3>
<p>From the metadata of sequences, we can plot, to understand in an overview, how is the succession of variants and sub-lineages of Omicrons in the US. We plot the frequencies of each category and the raw count numbers.</p>
<div class="screen-page">
<p><img src="fig/plot_us_raw_count_genomes.png" class="img-fluid"></p>
</div>
<p>After it, we break the data in a more grainular level, of states. And we plot the the counts of variants as well as the frequencies of them, over the whole US, for all states. This helps in visualizing the successive waves of Omicron sub-variants.</p>
<div class="screen-page">
<p><img src="fig/plot_raw_count_genomes_us.png" class="img-fluid"></p>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>